.github/workflows/ci.yaml

```diff
@@ -34,12 +36,72 @@ jobs:
             libvpx-dev \
             libx11-dev \
             libx264-dev \
-            libxext-dev
+            libxext-dev \
+            nasm \
+            yasm
+      
+      # Cache FFmpeg build - include both source and install directories
+      - name: Cache FFmpeg build
+        uses: actions/cache@v4
+        id: ffmpeg-cache
+        with:
+          path: |
+            pkg/codec/ffmpeg/tmp/n7.0
+          key: ffmpeg-linux-n7.0-${{ hashFiles('pkg/codec/ffmpeg/Makefile') }}
+          restore-keys: |
+            ffmpeg-linux-n7.0-
+      
+      - name: Check if FFmpeg libraries exist
+        id: ffmpeg-check
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          echo "=== Checking FFmpeg cache status ==="
+          if [ -f "tmp/n7.0/lib/libavcodec.a" ] && [ -f "tmp/n7.0/lib/pkgconfig/libavcodec.pc" ]; then
+            echo "FFmpeg libraries found in cache"
+            echo "ffmpeg_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "FFmpeg libraries missing or incomplete"
+            ls -la tmp/ 2>/dev/null || echo "tmp directory does not exist"
+            ls -la tmp/n7.0/ 2>/dev/null || echo "n7.0 directory does not exist"
+            ls -la tmp/n7.0/lib/ 2>/dev/null || echo "lib directory does not exist"
+            echo "ffmpeg_exists=false" >> $GITHUB_OUTPUT
+          fi
+      
+      - name: Build FFmpeg (if not cached or incomplete)
+        if: steps.ffmpeg-cache.outputs.cache-hit != 'true' || steps.ffmpeg-check.outputs.ffmpeg_exists != 'true'
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          echo "=== Starting FFmpeg build ==="
+          ls -la tmp/ 2>/dev/null || echo "tmp directory does not exist"
+          ls -la tmp/n7.0/ 2>/dev/null || echo "n7.0 directory does not exist"
+          make build-ffmpeg
+          echo "=== FFmpeg build finished ==="
+          ls -la tmp/n7.0/
+          ls -la tmp/n7.0/lib/ 2>/dev/null || echo "lib directory not found after build"
+          ls -la tmp/n7.0/include/ 2>/dev/null || echo "include directory not found after build"
+      
+      - name: Verify FFmpeg installation
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          ls -la tmp/n7.0/
+          ls -la tmp/n7.0/lib/ || echo "lib directory not found"
+          ls -la tmp/n7.0/include/ || echo "include directory not found"
+          pkg-config --exists --print-errors libavcodec || echo "pkg-config check failed"
+        env:
+          PKG_CONFIG_PATH: ${{ github.workspace }}/pkg/codec/ffmpeg/tmp/n7.0/lib/pkgconfig
+      
       - name: Run Test Suite
-        run: make test
+        run: |
+          make test \
+          && cd pkg/codec/ffmpeg \
+          && make test
```

Comment:
It might be better to separate the CI workflow for ffmpeg codec with path filter like
```yaml
name: CI
on:
  pull_request:
    branches:
      - master
    paths:
      - pkg/codec/ffmepg
  push:
    branches:
      - master
    paths:
      - pkg/codec/ffmepg
```

to make it easy to determine which of the main module and the ffmpeg codec module is failing.
Also, the main module build duration will not be affected by ffmpeg cache expiration by separating the workflow.